src/ucs/sys/sock.h

```diff
@@ -30,10 +30,24 @@ BEGIN_C_DECLS
 #define UCS_SOCKET_INET6_PORT(_addr) (((struct sockaddr_in6*)(_addr))->sin6_port)
 
 
-/* Returns an error if the default error handling should be
- * done (the error value will be returned to a caller),
- * otherwise `UCS_OK` */
-typedef ucs_status_t (*ucs_socket_io_err_cb_t)(void *arg, int io_errno);
+/**
+ * Error callback to handle errno and status of a given socket IO operation.
+ *
+ * @param [in] arg       User's argument for the error callback.
+ * @param [in] io_errno  Errno set for a given IO operation.
+ * @param [in] io_status Status set for a given IO operation.
+ *
+ * @return UCS_OK (UCS_ERR_CANCELED will be returned as a result of the IO
```

Comment:
not clear the relation between OK and CANCELED?

Reply:
when the callback returns UCS_OK, the IO operation (send/recv/sendv or thei non-block variants) will return UCS_ERR_CANCELED